pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -173,12 +173,14 @@ func (c *Reconciler) reconcilePA(ctx context.Context, rev *v1.Revision) error {
 	// We no longer require immutability, so need to reconcile PA each time.
 	tmpl := resources.MakePA(rev, deployment)
 	logger.Debugf("Desired PASpec: %#v", tmpl.Spec)
-	if !equality.Semantic.DeepEqual(tmpl.Spec, pa.Spec) {
-		diff, _ := kmp.SafeDiff(tmpl.Spec, pa.Spec) // Can't realistically fail on PASpec.
-		logger.Infof("PA %s needs reconciliation, diff(-want,+got):\n%s", pa.Name, diff)
+	if !equality.Semantic.DeepEqual(tmpl.Spec, pa.Spec) || !equality.Semantic.DeepEqual(tmpl.Annotations, pa.Annotations) {
```

Comment:
Hi @saileshd1402 I suppose you want to avoid creating a new revision, what is the use case you have? 
Could you describe the problem that you have?

Reply:
Hi @skonto, essentially we are facing issues with manually scaling/updating replicaset. By default when we change scale, knative-serving brings up the pods associated with the new revision, wait for them to be active then scales down the previous revision. This doesn't work for on prem scenarios since there are fixed limited resources especially when dealing with GPUs so the new revision never gets ready.

Reply:
Hi @saileshd1402 this is something https://github.com/knative-extensions/serving-progressive-rollout tries to address. You can ask @houshengbo who are facing the same in their org and maintains the extension.

Reply:
We did try using [progressive rollout](https://github.com/knative-extensions/serving-progressive-rollout) extension but it had it's own issues we faced: 
1. https://github.com/knative-extensions/serving-progressive-rollout/issues/200
2. https://github.com/knative-extensions/serving-progressive-rollout/issues/202
3. https://github.com/knative-extensions/serving-progressive-rollout/issues/203

In a brief summary, we have noticed that the "resourceUtil" strategy fails to do graceful traffic transfer during consecutive updates, particularly when resource limits are hit. This leads to stuck states and failed requests, as traffic may still continue to direct to terminated revisions while new pods remain in pending state.